### DISCUSSION TITLE
Question about ColorTemperature

### DISCUSSION THREAD

**Discussion by Chen-YanZhang** [HomeSpan/HomeSpan#427] [Q&A] [answered]
First of all, thank you for this wonderful library. I am planning using this library to build my own table lamp with color temperature control and brightness control funtion. For hardware i have 2 different color temperature LEDs: 2700K+5000K.

Here is my questions:

If i decalre a new Accessory as below. I can get a controller in Home app. 
 ```
  new Service::LightBulb();
  new Characteristic::On(); 
  new Characteristic::Brightness();
  new Characteristic::ColorTemperature();

```
But I don't know how to control the color temperature of the real device. There is no relevant code in DEV_LED.h. Suppose I use Pin 16, 17 as the Pwm output pins of 2700K and 5000K respectively. The control method of RGB-LED does not seem to be suitable for my situation. 

Does it mean that I need to write a DEV_LED.h myself to control the color temperature? An equation of two output pins and color temperature.
#

**Comment by HomeSpan** [comment, marked as answer]
There is no generic formula that I am aware of for converting color temperatures into a split between two fixed-temperature devices.  A lot of this is simply a matter of personal preference.  For example you can scale the PWM output of each pin from 0-100% depending on the color temp chosen. If the user choses a temp of 2700K or less, set the output for the pin connected to 2700K LED to 100% and the pin connected to the 5000K to 0%.  If the user sets a temperature somewhere in the middle, set both to 50%, or one to 75% and the other to 25% if closer to one temperature than the other, etc.  You'll have to experiment with your actual device to see what looks best.
